tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -42,15 +48,40 @@ public static void main(String[] args) throws Exception {
             /* parse args */
             String topicName = res.getString("topic");
             long numRecords = res.getLong("numRecords");
-            int recordSize = res.getInt("recordSize");
+            Integer recordSize = res.getInt("recordSize");
             int throughput = res.getInt("throughput");
             List<String> producerProps = res.getList("producerConfig");
             String producerConfig = res.getString("producerConfigFile");
+            String payloadFilePath = res.getString("payloadFile");
+
+            // since default value gets printed with the help text, we are escaping \n there and replacing it with correct value here.
+            String payloadDelimiter = res.getString("payloadDelimiter").equals("\\n") ? "\n" : res.getString("payloadDelimiter");
 
             if (producerProps == null && producerConfig == null) {
                 throw new ArgumentParserException("Either --producer-props or --producer.config must be specified.", parser);
             }
 
+            if ((recordSize == null && payloadFilePath == null) || (recordSize != null && payloadFilePath != null)) {
```

Comment:
you can validate this in the `ArgumentParser` by using `parser.addMutuallyExclusiveGroup()` and adding `--record-size` and `--payload-file` to the returned group and making the group `.required()`